sdk/src/testing.rs

```diff
@@ -1,53 +1,58 @@
+// Utility function to handle call error message formatting and call to abort syscall
+pub fn handle_assert_err(catch_unwind_res: std::thread::Result<()>) -> ! {
+    if catch_unwind_res.is_err() {
+        match catch_unwind_res.err() {
+            Some(err) => match err.downcast::<String>() {
+                Ok(panic_msg_box) => crate::vm::abort(
+                    fvm_shared::error::ExitCode::USR_ASSERTION_FAILED.value(),
+                    Some(panic_msg_box.as_str()),
+                ),
+                Err(_) => crate::vm::abort(
+                    fvm_shared::error::ExitCode::USR_ASSERTION_FAILED.value(),
+                    None,
+                ),
+            },
+            None => unreachable!(),
+        };
+    }
+}
+
 // Wrapper around the assert macro to have a hand on which exit code we want to give to our failed
 // assertion
 #[macro_export]
 macro_rules! assert {
-    ($cond:expr, $(,)?) => {{
+    ($cond:expr $(,)?) => ({
         let res = std::panic::catch_unwind(|| {
             core::assert!($cond);
         });
-        if res.is_err() {
-            let panic_msg = match res.err() {
-                Some(err) => match err.downcast::<String>() {
-                    Ok(panic_msg_box) => Some(panic_msg_box.as_str()),
-                    Err(err) => None,
-                },
-                None => unreachable!(),
-            };
-
-            $crate::vm::abort(
-                fvm_shared::error::ExitCode::USR_ASSERTION_FAILED.value(),
-                panic_msg,
-            );
-        }
+        $crate::testing::handle_assert_err(res);
```

Comment:
Is there any reason we need to assert than catch it? Can't we just re-implement `assert` and abort directly? The implementation should be pretty simple.